### PR TITLE
BUG: numpy.split on non-UTC changes original time (#14042)

### LINKED ISSUE #14042: numpy.split on non-UTC, tz-aware data undergoes UTC roundtrip
#### Code Sample, a copy-pastable example if possible

```
import pandas as pd
import numpy as np
indices = pd.date_range('2016-01-01 00:00:00+0200', freq='S', periods=10)
np.split(indices, indices_or_sections=[])
Out:
[DatetimeIndex(['2015-12-31 20:00:00+02:00', '2015-12-31 20:00:01+02:00',
               '2015-12-31 20:00:02+02:00', '2015-12-31 20:00:03+02:00',
               '2015-12-31 20:00:04+02:00', '2015-12-31 20:00:05+02:00',
               '2015-12-31 20:00:06+02:00', '2015-12-31 20:00:07+02:00',
               '2015-12-31 20:00:08+02:00', '2015-12-31 20:00:09+02:00'],
              dtype='datetime64[ns, pytz.FixedOffset(120)]', freq='S')]
```
#### Expected Output

```
[DatetimeIndex(['2016-01-01 00:00:00+02:00', '2016-01-01 00:00:01+02:00',
               '2016-01-01 00:00:02+02:00', '2016-01-01 00:00:03+02:00',
               '2016-01-01 00:00:04+02:00', '2016-01-01 00:00:05+02:00',
               '2016-01-01 00:00:06+02:00', '2016-01-01 00:00:07+02:00',
               '2016-01-01 00:00:08+02:00', '2016-01-01 00:00:09+02:00'],
              dtype='datetime64[ns, pytz.FixedOffset(120)]', freq='S')]
```
#### output of `pd.show_versions()`

```
In [16]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 4.2.0-38-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.1
nose: 1.3.7
pip: 8.1.2
setuptools: 25.2.0
Cython: 0.24.1
numpy: 1.11.1
scipy: 0.18.0
statsmodels: None
xarray: None
IPython: 1.2.1
sphinx: None
patsy: None
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: None
tables: 3.1.1
numexpr: 2.6.1
matplotlib: 1.3.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999
httplib2: 0.8
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
boto: 2.42.0
pandas_datareader: None
```

### PR DESCRIPTION
**Pull request by wooyekim**
Root cause: When a DatetimeIndex is created by __array_wrap__,
round-trip of timezone occured

Solution: Remove the timezone of the original index when creating
a DatetimeIndex via __array_wrap__ and apply the timezone later

- [x] closes #14042
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -325,6 +325,7 @@ Indexing
 - Fixes ``DataFrame.loc`` for setting with alignment and tz-aware ``DatetimeIndex`` (:issue:`16889`)
 - Avoids ``IndexError`` when passing an Index or Series to ``.iloc`` with older numpy (:issue:`17193`)
 - Allow unicode empty strings as placeholders in multilevel columns in Python 2 (:issue:`17099`)
+- Fixes bug for ``numpy.split`` on pandas dataframe with non-utc timezone (:issue:`14042`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: bug in numpy array_wrap with a pandas Series/Index.

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -550,7 +550,14 @@ def __array_wrap__(self, result, context=None):
 
         attrs = self._get_attributes_dict()
         attrs = self._maybe_update_attributes(attrs)
-        return Index(result, **attrs)
+        from pandas.core.indexes.datetimes import DatetimeIndex
```
Comment (jreback) [inline pandas/core/indexes/base.py]: add this

```
diff --git a/pandas/core/indexes/base.py b/pandas/core/indexes/base.py
index de6221987..865e92201 100644
--- a/pandas/core/indexes/base.py
+++ b/pandas/core/indexes/base.py
@@ -99,6 +99,9 @@ def _new_Index(cls, d):
     if issubclass(cls, ABCPeriodIndex):
         from pandas.core.indexes.period import _new_PeriodIndex
         return _new_PeriodIndex(cls, **d)
+    elif issubclass(cls, ABCDatetimeIndex):
+        from pandas.core.indexes.datetimes import _new_DatetimeIndex
+        return _new_DatetimeIndex(cls, **d)
     return cls.__new__(cls, **d)
``` 

then you can simply call
return _new_Index(result attrs)

File: pandas/tests/indexes/datetimes/test_datetime.py [hunk 3]
```diff
@@ -778,3 +778,11 @@ def test_slice_duplicate_monotonic(self):
         result = idx._maybe_cast_slice_bound('2017-01-01', 'left', 'loc')
         expected = Timestamp('2017-01-01')
         assert result == expected
+
+    def test_split_date_range_with_timezone(self):
+        # https://github.com/pandas-dev/pandas/issues/14042
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_datetime.py]: add a 1-liner about what you are testing here

File: pandas/tests/indexes/datetimes/test_datetime.py [hunk 4]
```diff
@@ -778,3 +778,11 @@ def test_slice_duplicate_monotonic(self):
...
+        idx = DatetimeIndex(['2016-01-01 00:00:00', '2016-01-02 00:00:00'],
+                            tz='Asia/Seoul')
+        split = np.split(idx, indices_or_sections=[])
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_datetime.py]: use an ``tm.assert_index_equal`` and test for each piece.

Also test using a ``Series`` (can be here is ok)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/datetimes/test_datetime.py (hunk 4 above)
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_datetime.py]: ideally we could exercise more numpy ufuncs for 1d reshaping ops here as well.

Comment (wooyekim) [inline pandas/tests/indexes/datetimes/test_datetime.py]: I found that Series does not suffer from this issue. But I'll leave the testcase.

File: pandas/core/indexes/base.py [hunk 5]
```diff
@@ -550,7 +550,13 @@ def __array_wrap__(self, result, context=None):
 
         attrs = self._get_attributes_dict()
         attrs = self._maybe_update_attributes(attrs)
-        return Index(result, **attrs)
+        from pandas.core.dtypes.generic import ABCDatetimeIndex
```
Comment (jreback) [inline pandas/core/indexes/base.py]: not sure if you saw my comment. pls move this logic to ``_new_Index`` (and the import can be at the top of the file)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you rebase / update

File: doc/source/whatsnew/v0.21.0.txt [hunk 6]
```diff
@@ -325,6 +325,7 @@ Indexing
 - Fixes ``DataFrame.loc`` for setting with alignment and tz-aware ``DatetimeIndex`` (:issue:`16889`)
 - Avoids ``IndexError`` when passing an Index or Series to ``.iloc`` with older numpy (:issue:`17193`)
 - Allow unicode empty strings as placeholders in multilevel columns in Python 2 (:issue:`17099`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: move to 0.21.1

**Review by jreback — CHANGES_REQUESTED** [review]
can you rebase

**Pull request closed without merging**